### PR TITLE
Add /add_job endpoint which takes a POST containing just a audio url

### PR DESCRIPTION
**Pull request by yoninachmany**
r @sandlerben 
- expects a json string which looks like this
  `{
  "audioURL": "https://sound.com/random.mp3"
  }`
- responds with a 200 OK for now
- added `audioData` struct that unmarshalls json; added `json` package

screen shot 2016-02-13 at 9 31 34 am

Run with command:
`curl -X POST -d "{\"audioURL\": \"https://sound.com/random.mp3\"}"  http://localhost:8080/add_job`

### CODE REVIEW

**Comment by sandlerben** [comment]
@anoshaminai (test)

File: main.go [hunk 1]
```diff
@@ -20,3 +22,19 @@ func helloHandler(w http.ResponseWriter, r *http.Request) {
 	args := mux.Vars(r)
 	fmt.Fprintf(w, "Hello %s!", args["name"])
 }
+
+func jobHandler(w http.ResponseWriter, r *http.Request) {
```
Comment (sandlerben) [inline main.go]: Can you add documentation to this function. [Here](https://golang.org/doc/effective_go.html#commentary) are some tips for writing Go documentation.

> Doc comments work best as complete sentences, which allow a wide variety of automated presentations. The first sentence should be a one-sentence summary that starts with the name being declared.

Also, can we name it something slightly more semantic? How about `initiateTranscriptionJobHandler`? Or is that too verbose?

File: main.go [hunk 2]
```diff
@@ -20,3 +22,19 @@ func helloHandler(w http.ResponseWriter, r *http.Request) {
...
+	if err := json.NewDecoder(r.Body).Decode(&d); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json; charset=utf-8")
+	json.NewEncoder(w).Encode(d)
+}
+
+type audioData struct {
```
Comment (sandlerben) [inline main.go]: I would rename this. How about `transcriptionJobData`?

File: main.go [hunk 3]
```diff
@@ -5,12 +5,14 @@ import (
 	"net/http"
 	_ "net/http/pprof" // import for side effects
 
+	"encoding/json"
```
Comment (sandlerben) [inline main.go]: I recommend you use goimports as your formatter so this gets formatted properly.

File: main.go [hunk 4]
```diff
@@ -20,3 +22,19 @@ func helloHandler(w http.ResponseWriter, r *http.Request) {
 	args := mux.Vars(r)
 	fmt.Fprintf(w, "Hello %s!", args["name"])
 }
+
+func jobHandler(w http.ResponseWriter, r *http.Request) {
+	var d audioData
```
Comment (sandlerben) [inline main.go]: Can we name this something more semantic? Like `jsonData`?

File: main.go [hunk 5]
```diff
@@ -30,3 +32,23 @@ func helloHandler(w http.ResponseWriter, r *http.Request) {
...
 	fmt.Fprintf(w, "Hello %s!", args["name"])
 }
+
+// initiateTranscriptionJobHandle takes a POST request containing a json object,
+// decodes it into an audioData struct, and returns the struct json-encoded.
+func initiateTranscriptionJobHandler(w http.ResponseWriter, r *http.Request) {
+	var jsonData transcriptionJobData
+
+	// unmarshal from the response body directly into our struct
+	if err := json.NewDecoder(r.Body).Decode(&jsonData); err != nil {
```
Comment (sandlerben) [inline main.go:42]: :+1: for using idiomatic Go error checking

File: main.go [hunk 6]
```diff
@@ -30,3 +32,23 @@ func helloHandler(w http.ResponseWriter, r *http.Request) {
...
+		http.Error(w, err.Error(), http.StatusInternalServerError)
```
Comment (sandlerben) [inline main.go]: Is this really indicative of an `StatusInternalServerError`? I think it's a client error.

File: main.go [hunk 7]
```diff
@@ -30,3 +32,23 @@ func helloHandler(w http.ResponseWriter, r *http.Request) {
...
+		return
+	}
+
+	// return the struct encoded as json
+	w.Header().Set("Content-Type", "application/json; charset=utf-8")
```
Comment (sandlerben) [inline main.go]: How come you are returning the struct as json? Shouldn't we just return like "accepted"?

File: main.go (hunk 6 above)
Comment (yoninachmany) [inline main.go]: good call - came from copying tutorial example.

How's `StatusBadRequest` from https://golang.org/pkg/net/http/#pkg-constants?

File: main.go (hunk 7 above)
Comment (yoninachmany) [inline main.go]: sure - again, came from tutorial example.

File: main.go (hunk 6 above)
Comment (sandlerben) [inline main.go]: perfect

**Comment by sandlerben** [comment]
Besides my comments, lgtm

**Pull request merged**